packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -503,6 +503,7 @@ export class OdspDocumentService implements IDocumentService {
             this._opsCache?.flushOps();
         }
         this._opsCache?.dispose();
+        this.clearJoinSessionTimer();
```

Comment:
These questions are likely for separate task / PR:

This looks fine as is, but given that we already cancel it on disconnect, I wonder if scheduleJoinSessionRefresh() should be called on establishing connection only?
Also (unrelated), should we prevent connectToDeltaStream() calls in some form while there is active connection?
I think that case, and the case of 2 clients (main & summarizer) are fine in terms of number of joinSessions of calls thanks to PromiseCache.addOrGet() doing proper deduping for us, but I'd still 2 parallel connectToDeltaStream() will result in refresh being cancelled if either of such connections disconnects, so system assumes only one connection at any time per driver instance implicitly.

The alternative design - we move such logic into connection flow / connection object, including timers. That way we keep 2 timer instances, but disconnect on one connection results in another one going. But we already have enough code that assumes singleton (see this.currentConnection).

Can you please follow up here?

Reply:
Let's create an issue for this so we are tracking it. @jatgarg

Reply:
Sorry I didn't get what you are suggesting.
So main and summarizer client both will have their own delta connection, however they will probably share same join session response cache. And they will both have their own timer.

Reply:
Yes. Creating an issue. Was just trying to understand what is the issue :) I talked to him and creating an issue now.

Reply:
Created here: https://dev.azure.com/fluidframework/internal/_workitems/edit/2250